packages/create-solana-dapp/lib/get-args.ts

```diff
@@ -1,103 +1,131 @@
-
 import { CreateWorkspaceOptions } from 'create-nx-workspace'
+import { PackageManager } from 'nx/src/utils/package-manager'
+import { anchorTemplates } from './anchor-templates'
+import { getAppInfo } from './get-app-info'
+import { GetArgsResult } from './get-args-result'
 import { getPresets } from './get-presets'
-
-import { validateProjectName } from './validate-project-name'
-
-// eslint-disable-next-line @typescript-eslint/no-var-requires
-const packageJson = require('../package.json')
-
-const anchorTemplateChoices = ['counter', 'hello-world']
-
-export const app = {
-  name: packageJson.name,
-  version: packageJson.version,
-}
-
-function comment(text: string) {
-  return `
-
- ${text}
-`
-}
-
-export async function getArgs(): Promise<any> {
-  const { presets, presetChoices } = getPresets(packageJson.version)
-
-  const result = program
-    .name(packageJson.name)
-    .version(packageJson.version)
-    .option('-n, --name <name>', comment(`Name of the workspace`))
-    .option('--app-name <name>', comment(`Name of the frontend project (default: web)`))
-    .option('-p, --preset <preset>', comment(`Preset to use (options: ${presetChoices.join(', ')})`))
-    .option('-d, --dry-run', comment(`Dry run (default: false)`))
-    .option('-a, --anchor', comment(`Include anchor in the project (default: true)`))
-    .option('--anchor-name <anchor-name>', comment(`Anchor project name (default: anchor)`))
-    .option('--anchor-template <anchor-template>', comment(`Anchor template (default: counter)`))
-    .option('-pm, --package-manager <package-manager>', comment(`Package manager to use (default: npm)`))
+import { getPrompts, getUiLibraries } from './get-prompts'
+
+export async function getArgs(argv: string[], pm: PackageManager = 'npm'): Promise<GetArgsResult> {
+  // Get app info from package.json
+  const app = getAppInfo()
+
+  // Get info about the presets
+  const { presets, presetValues } = getPresets(app.version)
+
+  // Get the result from the command line
+  const input = program
+    .name(app.name)
+    .version(app.version)
+    .argument('[name]', 'Name of the project (default: prompt)')
+    .option(
+      '-p, --preset <preset>',
+      help(`Preset to use (default: prompt, options: ${presetValues.join(', ')})`),
+      (value: string) => {
+        if (!presetValues.includes(value)) {
+          throw new Error(`Invalid preset: ${value}`)
+        }
+        return value
+      },
+    )
+    .option('--ui <ui-library>', help(`UI library to use (default: prompt)`))
+    .option(
+      '-a, --anchor <template>',
+      help(`Name of the Anchor template to use (default: prompt, set to none to disable)`),
+      (value: string) => {
+        if (!anchorTemplates.includes(value)) {
+          throw new Error(`Invalid anchor template: ${value}`)
+        }
+        return value
+      },
+    )
+    .option('--anchor-build', help(`Build the anchor project (default: false)`), false)
```

Comment:
This extra `false` was printing the extra line on the help menu.
Does not having it set here make the values to not be false by default? Do we need to use the anchor input normalizer to set this default=false?